src/ngraph/op/quantize.cpp

```diff
@@ -0,0 +1,100 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/quantize.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+op::Quantize::Quantize(shared_ptr<Node> input,
+                       shared_ptr<Node> scale,
+                       shared_ptr<Node> offset,
+                       const element::Type& type,
+                       const AxisSet& axes,
+                       RoundMode round_mode)
+
+    : Op("Quantize", check_single_output_args({input, scale, offset}))
+    , m_type(type)
+    , m_axes(axes)
+    , m_round_mode(round_mode)
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::Quantize::validate_and_infer_types()
+{
+    enum
+    {
+        INPUT,
+        SCALE,
+        OFFSET
+    };
+
+    set_output_size(1);
+    set_output_type(0, m_type, get_input_shape(INPUT));
+
+    // TODO: longer term we probably want quantized types
+    // 1) for type safety - so quantized types are not passed to non-quantized ops
+    // 2) to reflect quantized type min/max which can vary e.g. [-127, 127] for "scaled" int8
+    NODE_VALIDATION_ASSERT(this, (m_type == element::u8 || m_type == element::i8))
+        << "Output element type (" << m_type << ") must be an 8-bit integer";
+
+    NODE_VALIDATION_ASSERT(this, get_input_element_type(INPUT).is_real())
+        << "Input element type (" << get_input_element_type(INPUT)
+        << ") must be a floating point number";
+
+    NODE_VALIDATION_ASSERT(this, get_input_element_type(SCALE) == get_input_element_type(INPUT))
+        << "Scale element type (" << get_input_element_type(SCALE)
+        << ") must match input element type (" << get_input_element_type(INPUT) << ")";
+
+    NODE_VALIDATION_ASSERT(this, get_input_element_type(OFFSET) == m_type)
+        << "Offset element type (" << get_input_element_type(OFFSET)
+        << ") must match input element type (" << m_type << ")";
```

Comment:
I don't think that will work, although I think it might work to use double the value, assuming we allow each of min/max to fall either on the edge or center of scale region and nothing else.